tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/im2latex_evaluator.py

```diff
@@ -0,0 +1,306 @@
+
+
+class Im2latexEvaluator(BaseEvaluator):
+    def __init__(self, dataset, reader, preprocessing, postprocessing, metric_executor, launcher, model):
+        self.dataset = dataset
+        self.preprocessing_executor = preprocessing
+        self.postprocessing_executor = postprocessing
+        self.metric_executor = metric_executor
+        self.launcher = launcher
+        self.model = model
+        self.reader = reader
+        self._metrics_results = []
+
+    @classmethod
+    def from_configs(cls, config):
+        dataset_config = config['datasets'][0]
+        dataset = Dataset(dataset_config)
+        data_reader_config = dataset_config.get('reader', 'opencv_imread')
+        data_source = dataset_config['data_source']
+        if isinstance(data_reader_config, str):
+            reader = BaseReader.provide(data_reader_config, data_source)
+        elif isinstance(data_reader_config, dict):
+            reader = BaseReader.provide(data_reader_config['type'], data_source, data_reader_config)
+        else:
+            raise ConfigError('reader should be dict or string')
+        preprocessing = PreprocessingExecutor(dataset_config.get('preprocessing', []), dataset.name)
+        postprocessing = PostprocessingExecutor(dataset_config.get('postprocessing', []), dataset.name)
+        metrics_executor = MetricsExecutor(dataset_config['metrics'], dataset)
+        launcher = create_launcher(config['launchers'][0], delayed_model_loading=True)
+        meta = dataset.metadata
+        model = SequentialModel(
+            config.get('network_info', {}),
+            launcher,
+            config.get('_models', []),
+            meta,
+            config.get('_model_is_blob'),
+        )
+        return cls(dataset, reader, preprocessing, postprocessing, metrics_executor, launcher, model)
+
+    def process_dataset(self, stored_predictions, progress_reporter, *args, **kwargs):
+        self._annotations, self._predictions = [], []
+
+        if progress_reporter:
+            progress_reporter.reset(self.dataset.size)
+        self.dataset_meta = self.dataset.metadata
+        for batch_id, (dataset_indices, batch_annotation) in enumerate(self.dataset):
+
+            batch_identifiers = [annotation.identifier for annotation in batch_annotation]
+            batch_input = [self.reader(identifier=identifier) for identifier in batch_identifiers]
+            batch_input = self.preprocessing_executor.process(batch_input, batch_annotation)
+            batch_input, batch_meta = extract_image_representations(batch_input)
+            batch_prediction = self.model.predict(batch_identifiers, batch_input)
+            batch_prediction = [CharacterRecognitionPrediction(label=batch_prediction,
+                                                                 identifier=batch_annotation[0].identifier)]
+            self._annotations.append(batch_annotation)
+            self._predictions.append(batch_prediction)
```

Comment:
since you concatenate lists, possible extend will be more suitable here

Reply:
Changed to extend